packages/eslint-plugin-office-addins/src/rules/no-navigational-load.ts

```diff
@@ -0,0 +1,88 @@
+import { TSESTree } from "@typescript-eslint/experimental-utils";
+import {
+  Reference,
+  Scope,
+  Variable,
+} from "@typescript-eslint/experimental-utils/dist/ts-eslint-scope";
+import { isLoadFunction, getPropertyNameInLoad } from "../utils/utils";
+import { isGetFunction } from "../utils/getFunction";
+import { getPropertyType, PropertyType } from "../utils/propertiesType";
+
+export = {
+  name: "no-navigational-load",
+  meta: {
+    type: <"problem" | "suggestion" | "layout">"problem",
+    messages: {
+      navigationalLoad:
+        "Calling load on the navigation property '{{loadValue}}' slows down your add-in.",
+    },
+    docs: {
+      description:
+        "Calling load on a navigation property causes unneeded data to load and slows down your add-in.",
+      category: <
+        "Best Practices" | "Stylistic Issues" | "Variables" | "Possible Errors"
+      >"Best Practices",
+      recommended: <false | "error" | "warn">false,
+      url: "https://docs.microsoft.com/en-us/office/dev/add-ins/develop/application-specific-api-model#scalar-and-navigation-properties",
+    },
+    schema: [],
+  },
+  create: function (context: any) {
+    function trimPropertyName(propertyName: string): string {
```

Comment:
I'm not sure what the value of this is.  Seems like you only getting part of the supposed property and throwing away potentially good information.

It also looks like you are handling the property in two different ways (with the other being findPropetiesRead())

Reply:
Yeah, this Trim is indeed throwing away useful information. It is together with the idea of the undefined properties, in which I only look to the last properties if it is navigational or scalar.

The findPropetiesRead actually builds the property name when the property is read. This trim propertyName works when the property is loaded, not read. I will change the name to address this.

Reply:
I looked up what the '/' is used for in argument and apparently it's short had for some navigation properties (I didn't know that, but maybe you did).  So instead of range.format.font.load("color") someone could call range.load("format/font/color") . . . is that why you are checking just the last element?  If so that it makes a little more sense, though you might want to check the earlier part are actual navigational elements.

Also I'm wondering in multiple properties can't be asked for in a single load statement and what that might look like.  We should account for that.

Reply:
Yeah, I knew about this syntax using "/" and that's what I was accounting for.
I agree that checking if the middle properties are navigational is useful and I will be doing that.

I also don't know if the load can load multiple properties. I guess using the syntax of loading an object:
item.load({font: color, font: size} could be used to load multiple scalar properties, but I'm not sure.